pmd-apex/src/main/resources/category/apex/errorprone.xml

```diff
@@ -291,6 +291,40 @@ public void bar(Integer a, Integer b) {
     // empty!
   }
 }
+]]>
+        </example>
+    </rule>
+
+        <rule name="OverrideBothEqualsAndHashcode"
+          language="apex"
+          since="6.3.0"
+          message="Ensure you override both equals() and hashCode()"
+          class="net.sourceforge.pmd.lang.apex.rule.errorprone.OverrideBothEqualsAndHashcodeRule"
+          externalInfoUrl="${pmd.website.baseurl}/pmd_rules_apex_errorprone.html#overridebothequalsandhashcode">
+        <description>
+Override both public boolean Object.equals(Object other), and public int Object.hashCode(), or override neither.  Even if you are inheriting a hashCode() from a parent class, consider implementing hashCode and explicitly delegating to your superclass.
```

Comment:
We should adjust this description for apex. Apex is not java, it is only similar.
Please refer to https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/langCon_apex_collections_maps_keys_userdefined.htm?search_text=hashCode

The equals method looks like this in apex:

```java
public Boolean equals(Object other) { }
```

HashCode method looks like this:

```java
public Integer hashCode() { }
```

Note: Boolean instead of boolean, Integer instead of int.

Reply:
That means, you need to adjust the examples. The tests use "Integer" as the result type of hashCode. Does Apex differ between "Boolean" and "boolean"?
Either way, I'd still use "Boolean" in the tests to be consistent with the reference documentation. There is the system class "Boolean", which wraps the primitive datatype "Boolean": https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/langCon_apex_primitives.htm and https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_methods_system_boolean.htm#apex_methods_system_boolean

Reply:
Ah, good catch! Thanks. will update

Reply:
updated version number & primitive names